File: agents/opencode-4-py/src/opencode_4_py/models/project.py
"""Project models."""

from __future__ import annotations

from pydantic import BaseModel, Field
from typing import Optional, List, Literal, Dict, Any


class ProjectTime(BaseModel):
    """Project time information."""
    created: int
    updated: int
    initialized: Optional[int] = None


class ProjectIcon(BaseModel):
    """Project icon."""
    url: Optional[str] = None
    override: Optional[str] = None
    color: Optional[str] = None


class ProjectCommands(BaseModel):
    """Project commands."""
    start: Optional[str] = None


class Project(BaseModel):
    """Project information."""
    id: str
    worktree: str
    vcs: Optional[Literal["git"]] = None
    name: Optional[str] = None
    icon: Optional[ProjectIcon] = None
    commands: Optional[ProjectCommands] = None
    time: ProjectTime
    sandboxes: List[str] = Field(default_factory=list)


class ProjectUpdateRequest(BaseModel):
    """Request to update project."""
    name: Optional[str] = None
    icon: Optional[ProjectIcon] = None
    commands: Optional[ProjectCommands] = None

File: sdk/oc4j/src/main/java/ai/opencode/sdk/model/project/Project.java
package ai.opencode.sdk.model.project;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Project {
    private String id;
    
    @JsonProperty("worktree")
    private String worktree;
    
    @JsonProperty("vcs")
    private VcsType vcs;
    
    private String name;
    private ProjectIcon icon;
    private ProjectCommands commands;
    private ProjectTime time;
    
    @JsonProperty("sandboxes")
    private List<String> sandboxes;

    public enum VcsType {
        @JsonProperty("git")
        GIT
    }
}